File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> mp;

        mp[0] = 1;
        int prefix = 0;
        int count = 0;
        int target;

        for(int i = 0 ; i< nums.size(); i++){
            prefix += nums[i];
            target = prefix - k;

            if(mp.find(target)!=mp.end()){
                count+=mp[target];
            }

            mp[prefix]++;
        }

        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        mp = {}

        mp[0] = 1

        sum = 0
        count = 0

        for i in nums:
            sum += i

            if sum - k in mp:
                count+=mp[sum-k]
            
            mp[sum] = mp.get(sum,0)+1

        return count

        